packages/medusa/src/loaders/database.ts

```diff
@@ -54,12 +52,29 @@ export default async ({
       (configModule.projectConfig.database_logging || false),
   } as DataSourceOptions)
 
-  await dataSource.initialize()
+  try {
+    await dataSource.initialize()
+  } catch (err) {
+    // database name does not exist
+    if (err.code === "3D000") {
+      throw new Error(
+        `Specified database does not exist. Please create it and try again.\n${err.message}`
+      )
+    }
+  }
 
-  if (isSqlite) {
-    await dataSource.query(`PRAGMA foreign_keys = OFF`)
-    await dataSource.synchronize()
-    await dataSource.query(`PRAGMA foreign_keys = ON`)
+  // If migrations are not included in the config, we assume you are attempting to start the server
+  // Therefore, throw if the database is not migrated
+  if (!dataSource.migrations?.length) {
+    try {
+      await dataSource.query(`select * from migrations`)
+    } catch (err) {
+      if (err.code === "42P01") {
+        throw new Error(
+          `Migrations missing. Please run 'yarn migrate' and try again.`
```

Comment:
thought: Maybe better to be package manager agnostic in the error message - e.g. `medusa migrations run`